### PR TITLE
[MRG] Segmentation example

### PR DESCRIPTION
**Pull request by mrahim**
It uses the auditory dataset for T1 segmentation, and then displays the overlay of the tissues.

### CODE REVIEW

File: examples/segmentation_demos.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import os
+from pypreprocess.subject_data import SubjectData
+from pypreprocess.datasets import fetch_spm_auditory
+from pypreprocess.nipype_preproc_spm_utils import _do_subject_segment
+from pypreprocess.reporting.check_preprocessing import plot_segmentation
+
+
+OUTPUT_DIR = 'segmentation_demos_output'
+def _spm_auditory_factory():
```
Comment (AlexandreAbraham) [inline examples/segmentation_demos.py]: The name is misleading. This has nothing to do with the factory design pattern. Please use a less connoted word.

Comment (mrahim) [inline examples/segmentation_demos.py]: Okay.

Comment (AlexandreAbraham) [inline examples/segmentation_demos.py]: I won! I made a comment in a pypreprocess PR before it's merged! Youhou!

Comment (dohmatob) [inline examples/segmentation_demos.py]: Troll detected ;)

**Comment by dohmatob** [comment]
* Add title in the generated plot (see plot_segmentation doc).
* Show plotted figure
* An maybe print location of results

File: examples/segmentation_demos.py (hunk 1 above)
Comment (dohmatob) [inline examples/segmentation_demos.py]: Won't you rather have fun with this one https://github.com/neurospin/pypreprocess/issues/8

**Comment by dohmatob** [comment]
Any progress ?

**Comment by mrahim** [comment]
Well, I don't know if we let the output segmentation in the example folder.
And how about plot_segmentation behavior ?

File: examples/segmentation_demos.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+@author: Mehdi RAHIM
+"""
+
+import os
+from pypreprocess.subject_data import SubjectData
+from pypreprocess.datasets import fetch_spm_auditory
+from pypreprocess.nipype_preproc_spm_utils import _do_subject_segment
+from pypreprocess.reporting.check_preprocessing import plot_segmentation
+import matplotlib.pyplot as plt
```
Comment (dohmatob) [inline examples/segmentation_demos.py]: I'd do the ``matplotlib`` import just after ``os``, otherwise your plots won't show.

**Comment by dohmatob** [comment]
OK, merging.

**Pull request merged**